### ISSUE TITLE
MultipleNegativesRankingLoss

### ISSUE THREAD

**Issue by joaomsimoes** [UKPLab/sentence-transformers#1345]
Hi Nils, quick question, what is the best way to train with MNRL? For one query can I have a list of positive passages? Or should only have one passage? And can a query have many positive passages?

`example = InputExample(texts=('query1', 'pos1'))`
Or
`example = InputExample(texts=('query1', ['pos1, pos2]'))`

Thanks

**Comment by nreimers** [comment]
You can have only one positive

**Comment by joaomsimoes** [comment]
And it is good practice to show many positives to the same query?

(query1, pos1), (query1, pos2),...

**Comment by nreimers** [comment]
Yes, you can do that. But you must ensure that a batch does not include the same query twice. You can do that via a custom DataLoader. See the training examples from msmarco and paraphrases for different DataLoaders how to achieve this

**Comment by joaomsimoes** [comment]
Understood! Thanks 👍
